expression/builtin_time.go

```diff
@@ -4561,13 +4574,28 @@ func (b *builtinMakeTimeSig) Clone() builtinFunc {
 func (b *builtinMakeTimeSig) evalDuration(row chunk.Row) (types.Duration, bool, error) {
 	dur := types.ZeroDuration
 	dur.Fsp = types.MaxFsp
-	hour, isNull, err := b.args[0].EvalInt(b.ctx, row)
+	// evalInt is the function for args[0] and args[1] to evaluate int64 result
+	evalInt := func(arg Expression) (int64, bool, error) {
+		if arg.GetType().EvalType() == types.ETReal {
+			fRes, isNull, err := arg.EvalReal(b.ctx, row)
+			if isNull || err != nil {
```

Comment:
how about directly return `int64(fRes), isNull, errors.Trace(handleInvalidTimeError(b.ctx, err))` and remove the `if` branch?